partiql-parser/src/lalr/lexer.rs

```diff
@@ -1,152 +1,25 @@
-use partiql_source_map::location::{ByteOffset, BytePosition, LineAndCharPosition, LineOffset};
+use partiql_source_map::location::{ByteOffset, BytePosition, ToLocated};
+use std::borrow::Cow;
```

Comment:
I haven't seen Cow (clone-on-write) pointers before. In what situations are they used?

Reply:
For the usage here (basically either a `str` or a `String`), see https://jwilm.io/blog/from-str-to-cow/